### PR TITLE
Fix compiler warnings, part 7

### PR DESCRIPTION
**Pull request by nilason**
Fixes compiler warnings:

- -Winteger-overflow
- -Wlogical-not-parentheses
- -Wunused-value
- -Wwritable-strings
- -Wundefined-inline

Seventh part addressing #1247.

There are a couple of `-Wlogical-not-parentheses` warnings and I'm not absolutely sure how to correctly address them. See comments in code.

Modules / code parts directly affected:

- lib/display
- lib/imagery
- lib/raster3d
- imagery/i.evapo.pm
- raster/r.viewshed
- vector/v.generalize
- vector/v.in.pdal

### CODE REVIEW

File: lib/display/draw2.c [hunk 1]
```diff
@@ -124,6 +124,8 @@ static void reduce_path(struct path *dst, const struct path *src, double eps)
 	struct vertex *v1 = &src->vertices[i];
 	struct vertex *v2 = &src->vertices[i+1];
 
+	/* logical not is only applied to the left hand  */
+	/* side of this comparison '!v2->mode != P_MOVE' */
 	if (fabs(v1->x - v0->x) < eps && fabs(v1->y - v0->y) < eps &&
 	    fabs(v1->x - v2->x) < eps && fabs(v1->y - v2->y) < eps &&
 	    v0->mode != P_MOVE && v1->mode != P_MOVE && !v2->mode != P_MOVE)
```
Comment (metzm) [inline lib/display/draw2.c:131]:
```suggestion
	    v0->mode != P_MOVE && v1->mode != P_MOVE && (!v2->mode) != P_MOVE)
```

File: lib/imagery/list_subgp.c [hunk 2]
```diff
@@ -34,6 +34,8 @@ char **list_subgroups(char *group, const char *mapset, int *subgs_num)
     sprintf(buf, "group/%s/subgroup", group);
     G_file_name(path, buf, "", mapset);
 
+    /* logical not is only applied to the left hand side of */
+    /* this comparison '!G_lstat(path, &sb) == 0'           */
     if (!G_lstat(path, &sb) == 0 || !S_ISDIR(sb.st_mode))
```
Comment (metzm) [inline lib/imagery/list_subgp.c:39]:
```suggestion
    if ((!G_lstat(path, &sb)) == 0 || !S_ISDIR(sb.st_mode))
```

File: lib/raster3d/close.c [hunk 3]
```diff
@@ -106,6 +106,8 @@ static int close_cell_new(RASTER3D_Map * map)
 	return 0;
     }
 
+    /* logical not is only applied to the left hand   */
+    /* side of this comparison '!close_new(map) != 0' */
     if (!close_new(map) != 0) {
```
Comment (metzm) [inline lib/raster3d/close.c:111]:
```suggestion
    if (!close_new(map)) {
```

File: lib/raster3d/close.c [hunk 4]
```diff
@@ -126,6 +128,8 @@ static int close_old(RASTER3D_Map * map)
 
 static int close_cell_old(RASTER3D_Map * map)
 {
+    /* logical not is only applied to the left hand   */
+    /* side of this comparison '!close_old(map) != 0' */
     if (!close_old(map) != 0) {
```
Comment (metzm) [inline lib/raster3d/close.c:133]:
```suggestion
    if (!close_old(map)) {
```

File: lib/raster3d/close.c [hunk 5]
```diff
@@ -155,6 +159,8 @@ int Rast3d_close(RASTER3D_Map * map)
 	}
     }
     else {
+	/* logical not is only applied to the left hand side */
+	/* of this comparison '!close_cell_old(map) != 0'    */
 	if (!close_cell_old(map) != 0) {
```
Comment (metzm) [inline lib/raster3d/close.c:164]:
```suggestion
	if (!close_cell_old(map)) {
```

**Comment by nilason** [comment]
Thanks! Implemented your suggestions and reworded commit messages.

**Comment by metzm** [comment]
> Thanks! Implemented your suggestions and reworded commit messages.

Can you please resolve the conversations where you applied my suggestions? That makes the next round of review easier. Thanks!

**Comment by nilason** [comment]
> Can you please resolve the conversations where you applied my suggestions? That makes the next round of review easier. Thanks!

Ok, I will do so.

**Comment by nilason** [comment]
Ready, will merge on approval.

**Review by metzm — APPROVED** [review]
Ready to merge, thanks a lot for this PR!

**Comment by nilason** [comment]
> Ready to merge, thanks a lot for this PR!

Merged. Thank you!

**Pull request closed without merging**